ygot/path_types.go

```diff
@@ -47,9 +54,38 @@ type NodePath struct {
 	p             PathStruct
 }
 
-// ResolvePath is a helper which returns the root PathStruct and absolute path
-// of a PathStruct node.
-func ResolvePath(n PathStruct) (PathStruct, []*gpb.PathElem, []error) {
+// FakeRootPathStruct is an interface that is implemented by the fake root path
+// struct type.
+type FakeRootPathStruct interface {
+	PathStruct
+	GetId() string
```

Comment:
If you don't need to export CustomData, could change the method names to the cleaner "Id()" and "CustomData()." The "Get" prefixed is sometimes frowned on as a "stop word" in Go.